File: Java/Livros/Fundamentos da programação de computadores/Capítulo 6 Vetor/Ex1.java
/* Faça um programa que preencha um vetor com nove números inteiros, calcule e mostre os números
primos e suas respectivas posições. */

import java.util.Random;

public class Ex1 {
    static public void main(String[] args) {
        // declara vetor de 9 posições
        int[] Vetor = new int[9];

        // cria objeto Random
        Random rand = new Random();

        // loop principal
        // preenche vetor com numeros aleatorio
        for (int i = 0; i < Vetor.length; i++)
            Vetor[i] = rand.nextInt(100);

        //verifica numero e posição no vetor 
        for (int i = 0; i < Vetor.length; i++) {
            if (Primo(Vetor[i]) == true)
                System.out.println(" Numero: " + Vetor[i] + " Posicao: " + i + " ");
        }

    }

    // verifica c é primo
    static public boolean Primo(int num) {
        {
            if (num <= 1) {
                return false;
            }
            for (int i = 2; i <= num / 2; i++) {
                if ((num % i) == 0)
                    return false;
            }
            return true;
        }

    }

}

File: Estrutura de Dados I/Livros/Fundamentos da Programação de Computadores/6. Vetor/Exercícios Resolvidos/Ex1.c
/*
Faça um programa que preencha um vetor com nove números inteiros, calcule e mostre os números
primos e suas respectivas posições
*/
#include <stdio.h>
#include <math.h>

void leNum(int num[]);
void ehPrimo(int num[]);

int main()
{
    int num[9];

    leNum(num);

    ehPrimo(num);

    return 0;

} // end main

void leNum(int num[])
{
    // loop principal
    for (int i = 0; i < 9; i++)
    {
        scanf("%d", &num[i]);
    } // end for

} // end leNum

void ehPrimo(int num[])
{
    int cont;

    // percorre vetor
    for (int i = 0; i < 9; i++)
    {
        cont = 0;

        // trata primo
        for (int j = 2; j <= sqrt(num[i]); j++)
        {
            if (num[i] % j == 0)
                cont = 1;
        } // end for

        if (cont == 0)
            printf("%d %d", num[i], i);

    } // end for

} // end ehPrimo